src/server.cpp

```diff
@@ -383,8 +384,14 @@ int Server::start(int argc, char** argv) {
         CLIParser parser;
         ServerSettingsImpl serverSettings;
         ModelsSettingsImpl modelsSettings;
+        HfDownloader hfDownloader;
         parser.parse(argc, argv);
-        parser.prepare(&serverSettings, &modelsSettings);
+        parser.prepare(&serverSettings, &modelsSettings, &hfDownloader);
+
+        if (hfDownloader.pull_hf_model) {
```

Comment:
Shouldn't this be responsibility of ServableModule?

Reply:
I think we should separate the server mode and hf download mode.